internal/sync/apiCollection.go

```diff
@@ -123,6 +125,36 @@ func (a *apiCollector) PendingBlock() *feeder.StarknetBlock {
 	return a.pendingBlock
 }
 
+// fetchContractCode fetch the code of the contract from the Feeder Gateway.
+func (a *apiCollector) fetchContractCode(stateDiff *types.StateDiff) *CollectorDiff {
+	collectedDiff := &CollectorDiff{
+		stateDiff: stateDiff,
+		Code:      make(map[string]*types.Contract),
+	}
+	for _, deployedContract := range stateDiff.DeployedContracts {
+
+		contractFromApi, err := a.client.GetFullContractRaw(deployedContract.Address.Hex0x(), "",
+			strconv.FormatInt(stateDiff.BlockNumber, 10))
+		if err != nil {
+			a.logger.With("Block Number", stateDiff.BlockNumber,
+				"Contract Address", deployedContract.Address.Hex0x()).
+				Error("Error getting full contract")
+			return collectedDiff
+		}
+
+		contract := new(types.Contract)
+		err = contract.UnmarshalRaw(contractFromApi)
+		if err != nil {
+			a.logger.With("Block Number", stateDiff.BlockNumber,
+				"Contract Address", deployedContract.Address.Hex0x()).
+				Error("Error unmarshalling contract")
```

Comment:
We can reduce scope and improve readability here as follows.

```suggestion
		if err := contract.UnmarshalRaw(contractFromApi); err != nil {
			a.logger.With(
				"Block Number", stateDiff.BlockNumber,
				"Contract Address", deployedContract.Address.Hex0x(),
			).Error("Error unmarshalling contract")
```

Reply:
done